pkg/urbit/vere/io/term.c

```diff
@@ -1424,20 +1430,49 @@ _term_ef_blit(u3_utty* uty_u,
     } break;
 
     case c3__url: {
-      u3_noun txt = u3t(blt);
+      //  platform-agnostically opening the default web browser from within a
+      //  c program is an unsolved problem.
+    } break;
```

Comment:
This is actually a little bit complicated. Printing it would give it the same semantics that `%put` has, requiring the sender to be aware of where the cursor is, make sure the url isn't overwriting anything important on-screen, and/or print a newline to make space for it.

Maybe that's fine, and we want to treat it that way because we can't guarantee the url will be opened. Thinking about the ideal future where all clients support all the blits fully, that's not what we want `%url` to do though.

The alternative is pushing responsibility for visually printing the url to whoever is emitting the `%url`.

Reply:
Looking into this again, perhaps we can get away with a simple `system()` depending on which OS we're building for. `open $url` on MacOS, `xdg-open $url` on Linux, `start $url` (or similar) on Windows.

But maybe there's some security concern there around apps being able to both write to disk (`%sav`) and open arbitrary strings.

Reply:
@philipcmonk the thing I mentioned during our last call, we actually already do:

https://github.com/urbit/urbit/blob/d138ab254b9985a1fdf50cdbdcf78e6f7fd1479e/pkg/arvo/lib/hood/drum.hoon#L916

That is, I was going to update drum to always print the url, in addition to sending a `%url` blit. Turns out I already did this five(!) months ago in 0d8288e.